File: TypeScript/Factorial.ts
import * as readline from 'readline'

const readInterface = readline.createInterface({
    input: process.stdin,
    output: process.stdout
})

const Factorial=(n:number)=>{

    if(n < 0){
        throw new Error('1以上の整数を入力してください')
    }
    let result:number
    if(n == 1){
        result = 1
    }else{
        result = n * Factorial(n - 1)
    }

    return result
}

readInterface.question('整数を入力してください',(n:string)=>{
    readInterface.close()

    const N = Number(n)

    try{
        const result = Factorial(N)
        console.log(`${N}の階乗は${result}です`)
    }
    catch(e){
        console.log(e)
    }
})

File: Python/factorial.py

def factorial(n:int)->int:
    """
    与えられた整数nの階乗(n!)を求める

    Parameters
    ----------
    n: int
        対象整数

    Returns
    -------
    result: int
        nの階乗
    """
    if type(n) is not int:
        raise TypeError("引数nの型がint型ではありません")
    elif n < 0:
        raise ValueError("1以上の整数を入力してください")

    if n == 1:
        result = 1
    else:
        result = n * factorial(n -1)

    return result


if __name__ == '__main__':

    n = int(input('整数を入力してください'))
    # n='a'

    try:

        result = factorial(n)
        print('{}の階乗は{}です'.format(n, result))

    except TypeError as e:
        print(e)
    except ValueError as e:
        print(e)